drivers/soundwire/intel.c

```diff
@@ -1051,6 +1051,21 @@ static int intel_trigger(struct snd_pcm_substream *substream, int cmd, struct sn
 	return ret;
 }
 
+static int intel_component_probe(struct snd_soc_component *component)
+{
+	int ret;
+
+	/*
+	 * make sure the device is pm_runtime_active before initiating
+	 * bus transactions during the card registration
+	 */
+	ret = pm_runtime_resume(component->dev);
```

Comment:
You use `pm_runtime_resume` while the commit message says `pm_runtime_resume_and_get()`. Do you want to use `pm_runtime_resume_and_get()` here and codec side?

Reply:
Thanks @bardliao, I should have added an important point: in this case, we could do a resume_and_get but we would have to do a put_autosuspend immediately, so the sequence can be degraded to resume only. There's no real benefit in taking a reference to release it immediately.

One drawback is that this doesn't guarantee that the bus will remain on for the remainder of the card registration, but at least it will guarantee that the bus is started when the card registration begins. Other parts such as the codec probe should also help make sure the bus remains on.

do you mind if we do this with a fixup so that we don't require a complete review?

Reply:
sure